### PR TITLE
[Fix #3406] Avoid publishing internal variables

### LINKED ISSUE #3406: Internal variables changes should not be published externally
### Description

Variable marked as internal by metadata should not be published to external systems 

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by fjtirado**
Fixes https://github.com/apache/incubator-kie-kogito-runtimes/issues/3406

### CODE REVIEW

File: api/kogito-events-core/src/main/java/org/kie/kogito/event/impl/ProcessInstanceEventBatch.java [hunk 1]
```diff
@@ -120,7 +120,9 @@ private void addDataEvent(ProcessEvent event) {
     }
 
     private void handleProcessVariableEvent(ProcessVariableChangedEvent event) {
-
+        if (event.getTags().contains("internal")) {
```
Comment (nmirasch) [inline api/kogito-events-core/src/main/java/org/kie/kogito/event/impl/ProcessInstanceEventBatch.java:123]: @fjtirado it wont be better to use the existing constant in bpmn variables?

**Review by wmedvede — APPROVED** [review]
agree with @nmirasch

File: api/kogito-events-core/src/main/java/org/kie/kogito/event/impl/ProcessInstanceEventBatch.java (hunk 1 above)
Comment (fjtirado) [reply api/kogito-events-core/src/main/java/org/kie/kogito/event/impl/ProcessInstanceEventBatch.java:123]: good catch! done

**Review by nmirasch — APPROVED** [review]
thanks!

**Pull request merged**